algorithms.py

```diff
@@ -771,6 +771,12 @@ def flush_edges(self):
             self.tables.edges.add_row(left, right, parent, child)
             self.edge_buffer = []
 
+    def update_node_flag(self, node_id, flag):
+        self.flush_edges()
```

Comment:
Do we need this flush_edges? Should probably be done in the other function we just added?

Reply:
Adding `flush_edges()` to `store_additional_nodes_edges()` would flush the edges twice in some cases, as it is also called by `store_node()`.

Reply:
I see - should we call flush_edges to the non-store-node path in store_additional_nodes_edges then?

Reply:
It's a surprising side-effect to have here in ``update_node_flag``